rebar.config

```diff
@@ -34,7 +34,8 @@
                    ]},
             {prod, [{relx, [{dev_mode, false},
                             {include_erts, true},
-                            {include_src, false}]}]
+                            {include_src, false}]},
+                    {erl_opts, [warnings_as_errors, true]}]
```

Comment:
all profiles should have warnings as errors. The sooner we spot it, the better.

Reply:
Test profile can't have, because of `-compile(export_all).`.
For dev1 and local I would say it's questionable. User sees the warning always when compiling so user is informed and we make sure it's not missed by including it in "prod" so Travis will fail.

Sometimes when developing one wants to add a function and not use it at first. Or one wants to add a stub function that ignores some of it's parameters. Such check would prevent one from running dialyzer without fixing warnings first or doing any manual tests.

Reply:
For the test profile, at least in our own code, we should use
```erlang
-compile([export_all, nowarn_export_all]).
```